### ISSUE TITLE
filter reset if the contents of a directory is changed

### ISSUE THREAD

**Issue by leorosa** [jarun/nnn#199]
While browsing my `~/Downloads/` directory, I noticed that any filter that I tried to set (calling `/`) was lost after a few seconds, and all files were shown again. 

It occurred because a file was being downloaded: its size varied with time. With more testing, I've seen that this happen every time a directory content changes: if a file is created or renamed outside of nnn, the filter is lost.

This behavior may be annoying. Can it be adjusted, filtering the contents of a directory even if a file changes?

Steps to reproduce:
1. `sleep 10 && ls > testFile &`
2. `nnn`
3. filter the contents, e.g. by typing `/filter`
4. wait 10 seconds; all files are shown again

**Comment by jarun** [comment]
Use the _navigate-as-you-type_ mode where this is blocked.

**Comment by jarun** [comment]
With a temporary filter, the state is lost when the contents of the directory change.

**Comment by jarun** [comment]
Even with tmp filter mode this should not happen.

**Comment by jarun** [comment]
I think after step 3, you are pressing up/down arrow keys which exits the filter mode but the content won't change immediately. However, if there is a change in the dir all content will be refreshed.

**Comment by leorosa** [comment]
You're right, I was pressing the arrow keys. In fact, even if I press them after step 4, the content is refreshed.
I'll check the _navigate-as-you-type_ mode. Somehow, I never used it...
